src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -142,14 +150,14 @@ class DirectStepI : public StepImp<IOUAmount, IOUAmount, DirectStepI>
     void setCacheLimiting (
         IOUAmount const& fwdIn,
         IOUAmount const& fwdSrcToDst,
-        IOUAmount const& fwdOut);
+        IOUAmount const& fwdOut,
+        bool srcRedeems);
 
     friend bool operator==(DirectStepI const& lhs, DirectStepI const& rhs)
     {
         return lhs.src_ == rhs.src_ &&
             lhs.dst_ == rhs.dst_ &&
-            lhs.currency_ == rhs.currency_ &&
-            lhs.noTransferFee_ == rhs.noTransferFee_;
+            lhs.currency_ == rhs.currency_;
```

Comment:
I want a direct step where Alice sends Bob $1/GW to be equal to another direct step that does the same thing, regardless of the context the step appears in the strand. If we want to know if a collection of steps are equal, we compare the collection.